recipes/libsamplerate/all/conanfile.py

```diff
@@ -25,53 +30,62 @@ class LibsamplerateConan(ConanFile):
         "fPIC": True,
     }
 
-    exports_sources = "CMakeLists.txt"
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
```

Comment:
@SpaceIm it should be covered by try..except too. When building on Windows and shared library, with Conan v2, fPIC will no longer exist at this point, because it was removed by config_options, then, the recipe will fail when trying to remove fPIC again here.

Reply:
Honestly, it's really ugly & confusing to protect all these deletions. I would prefer an elegant way to say that fPIC doesn't matter on Windows, and libcxx/cppstd for C libraries.

Reply:
Me too. It deserves a new feature.

Reply:
I opened a new feature request to discuss new ideas: https://github.com/conan-io/conan/issues/11983